tensorboard/webapp/metrics/effects/index.ts

```diff
@@ -320,11 +337,42 @@ export class MetricsEffects implements OnInitEffects {
     ofType(actions.metricsClearAllPinnedCards),
     withLatestFrom(
       this.store.select(selectors.getEnableGlobalPins),
-      this.store.select(selectors.getShouldPersistSettings)
+      this.store.select(selectors.getShouldPersistSettings),
+      this.store.select(selectors.getMetricsSavingPinsEnabled)
     ),
     filter(
-      ([, enableGlobalPins, shouldPersistSettings]) =>
-        enableGlobalPins && shouldPersistSettings
+      ([
+        ,
+        enableGlobalPinsFeature,
+        shouldPersistSettings,
+        isMetricsSavingPinsEnabled,
+      ]) =>
+        enableGlobalPinsFeature &&
+        shouldPersistSettings &&
+        isMetricsSavingPinsEnabled
+    ),
+    tap(() => {
+      this.savedPinsDataSource.removeAllScalarPins();
+    })
+  );
+
+  private readonly disableSavingPins$ = this.actions$.pipe(
```

Comment:
I was a bit confused by the naming at first because the effect doesn't actually disable (disabling is done elsewhere), this just removes pins.

WDYT of a naming that describes the actual action, like removeOnDisable?

Reply:
I changed to `removeSavedPinsOnDisable` 👍